wagtail/admin/edit_handlers.py

```diff
@@ -531,7 +531,7 @@ def db_field(self):
 
     def on_form_bound(self):
         self.bound_field = self.form[self.field_name]
-        self.heading = self.bound_field.label
+        self.heading = self.heading or self.bound_field.label
```

Comment:
The changes look good to me.
Would be great to update the docs as well

Reply:
Good point, thanks! I've added to the panels reference in latest commit